### PR TITLE
Remove old AgencyComm

### PR DESCRIPTION
**Pull request by maierlars**
Removed old AgencyCommManager and AgencyComm. Stub remains in place until all code places are ready for futures.

### CODE REVIEW

**Comment by maierlars** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/7913/

**Comment by maierlars** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/7918/

**Comment by maierlars** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/7931/

**Comment by maierlars** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/7963/

File: arangod/Cluster/ClusterFeature.cpp [hunk 1]
```diff
@@ -104,7 +105,8 @@ void ClusterFeature::collectOptions(std::shared_ptr<ProgramOptions> options) {
 
   options->addOption("--cluster.agency-prefix", "agency prefix",
                      new StringParameter(&_agencyPrefix),
-                     arangodb::options::makeFlags(arangodb::options::Flags::Hidden));
+                     makeFlags(arangodb::options::Flags::Hidden,
+                               arangodb::options::Flags::Obsolete));
```
Comment (jsteemann) [inline arangod/Cluster/ClusterFeature.cpp]: We need to document that the option becomes obsolete in 3.7, in both CHANGELOG and the upgrading-changes document for 3.7.
If the option was obsolete even before 3.7, we should document this as well.

File: arangod/Cluster/ClusterFeature.cpp [hunk 2]
```diff
@@ -299,20 +312,8 @@ void ClusterFeature::validateOptions(std::shared_ptr<ProgramOptions> options) {
...
-  size_t found = _agencyPrefix.find_first_not_of(
-      "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789/");
-
-  if (found != std::string::npos || _agencyPrefix.empty()) {
-    LOG_TOPIC("7259b", FATAL, arangodb::Logger::CLUSTER)
-        << "invalid value specified for --cluster.agency-prefix";
-    FATAL_ERROR_EXIT();
-  }
+  // changing agency namespace no longer needed
+  _agencyPrefix = "arango";
```
Comment (jsteemann) [inline arangod/Cluster/ClusterFeature.cpp]: do we need to support cases in which end users have adjusted the agency prefix themselves?
is this a concern?

File: arangod/Cluster/FollowerInfo.cpp [hunk 3]
```diff
@@ -105,11 +105,6 @@ Result FollowerInfo::add(ServerID const& sid) {
         _failoverCandidates = nextCandidates;  // will cast to std::vector<ServerID> const
       }
     }
-#ifdef DEBUG_SYNC_REPLICATION
```
Comment (jsteemann) [inline arangod/Cluster/FollowerInfo.cpp]: if `DEBUG_SYNC_REPLICATION` is removed here, do we still need to handle it in our CMakeLists.txt file?
Or can it be removed from there as well?

File: tests/Cluster/ClusterRepairsTest.cpp [hunk 4]
```diff
@@ -234,17 +234,17 @@ class ClusterRepairsTestBrokenDistribution
       public tests::LogSuppressor<Logger::FIXME, LogLevel::FATAL> {
  protected:
   // save old manager (may be null)
-  std::unique_ptr<AgencyCommManager> oldManager;
+  //std::unique_ptr<AgencyCommManager> oldManager;
```
Comment (jsteemann) [inline tests/Cluster/ClusterRepairsTest.cpp]: can all occurrences of `oldManager` be removed from the test new?

File: arangod/Cluster/ClusterFeature.cpp (hunk 2 above)
Comment (maierlars) [reply arangod/Cluster/ClusterFeature.cpp]: No, this does not even work, because other places have "arango" hardcoded. I marked this options as obsolete.

Comment (jsteemann) [reply arangod/Cluster/ClusterFeature.cpp]: Fine, then we don't need to support it, but just mention it in the CHANGELOG plus the "upgrading-changes" document in the docs repository.

File: arangod/Cluster/FollowerInfo.cpp (hunk 3 above)
Comment (maierlars) [reply arangod/Cluster/FollowerInfo.cpp]: I think there are other places as well.

Comment (jsteemann) [reply arangod/Cluster/FollowerInfo.cpp]: fine

**Comment by maierlars** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/7989/

File: arangod/Cluster/RestClusterHandler.cpp [hunk 5]
```diff
@@ -33,6 +33,7 @@
 #include "Replication/ReplicationFeature.h"
 #include "Rest/Version.h"
 
+#include <Agency/AsyncAgencyComm.h>
```
Comment (goedderz) [inline arangod/Cluster/RestClusterHandler.cpp]:
```suggestion
#include "Agency/AsyncAgencyComm.h"
```
(and move up)

File: arangod/Cluster/ServerState.cpp [hunk 6]
```diff
@@ -62,8 +62,10 @@ using namespace arangodb::basics;
 namespace {
 // whenever the format of the generated UUIDs changes, please make sure to
 // adjust this regex too!
-std::regex const uuidRegex("^(SNGL|CRDN|PRMR|AGNT)-[a-f0-9]{8}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{12}$");
-}
+std::regex const uuidRegex(
+    "^(SNGL|CRDN|PRMR|AGNT)-[a-f0-9]{8}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}-[a-"
+    "f0-9]{12}$");
```
Comment (goedderz) [inline arangod/Cluster/ServerState.cpp]: While a line break is okay, the actual column is unfortunate.

File: arangod/Cluster/v8-cluster.cpp [hunk 7]
```diff
@@ -23,6 +23,7 @@
 
 #include "v8-cluster.h"
 
+#include <Agency/AsyncAgencyComm.h>
```
Comment (goedderz) [inline arangod/Cluster/v8-cluster.cpp]:
```suggestion
#include "Agency/AsyncAgencyComm.h"
```

File: arangod/Cluster/v8-cluster.cpp [hunk 8]
```diff
@@ -487,7 +488,7 @@ static void JS_EndpointsAgency(v8::FunctionCallbackInfo<v8::Value> const& args)
     TRI_V8_THROW_EXCEPTION_USAGE("endpoints()");
   }
 
-  std::vector<std::string> endpoints = AgencyCommManager::MANAGER->endpoints();
+  std::deque<std::string> endpoints = AsyncAgencyCommManager::INSTANCE->endpoints();
```
Comment (goedderz) [inline arangod/Cluster/v8-cluster.cpp]:
```suggestion
  auto endpoints = AsyncAgencyCommManager::INSTANCE->endpoints();
```

File: arangod/RestHandler/RestShutdownHandler.cpp [hunk 9]
```diff
@@ -23,6 +23,7 @@
 
 #include "RestShutdownHandler.h"
 
+#include <Agency/AsyncAgencyComm.h>
```
Comment (goedderz) [inline arangod/RestHandler/RestShutdownHandler.cpp]:
```suggestion
#include "Agency/AsyncAgencyComm.h"
```

File: arangod/RestHandler/RestStatusHandler.cpp [hunk 10]
```diff
@@ -36,6 +36,7 @@
 #include <unistd.h>
 #endif
 
+#include <Agency/AsyncAgencyComm.h>
```
Comment (goedderz) [inline arangod/RestHandler/RestStatusHandler.cpp]:
```suggestion
#include "Agency/AsyncAgencyComm.h"
```

File: arangod/RestHandler/RestStatusHandler.cpp [hunk 11]
```diff
@@ -147,7 +148,7 @@ RestStatus RestStatusHandler::execute() {
         result.add("agencyComm", VPackValue(VPackValueType::Object));
         result.add("endpoints", VPackValue(VPackValueType::Array));
 
-        for (auto ep : manager->endpoints()) {
+        for (const auto& ep : manager->endpoints()) {
```
Comment (goedderz) [inline arangod/RestHandler/RestStatusHandler.cpp]:
```suggestion
        for (auto const& ep : manager->endpoints()) {
```

File: arangod/RestServer/BootstrapFeature.cpp [hunk 12]
```diff
@@ -21,6 +21,7 @@
 ////////////////////////////////////////////////////////////////////////////////
 
 #include "RestServer/BootstrapFeature.h"
+#include <Agency/AsyncAgencyComm.h>
```
Comment (goedderz) [inline arangod/RestServer/BootstrapFeature.cpp]:
```suggestion
#include "Agency/AsyncAgencyComm.h"
```

File: tests/AsyncAgencyComm/AsyncAgencyCommTest.cpp [hunk 13]
```diff
@@ -110,10 +112,10 @@ struct AsyncAgencyCommPoolMock final : public network::ConnectionPool {
   };
 
   struct Connection final : public fuerte::Connection {
-    Connection(AsyncAgencyCommPoolMock* mock, std::string const& endpoint)
+    Connection(AsyncAgencyCommPoolMock* mock, std::string  endpoint)
```
Comment (goedderz) [inline tests/AsyncAgencyComm/AsyncAgencyCommTest.cpp]:
```suggestion
    Connection(AsyncAgencyCommPoolMock* mock, std::string endpoint)
```

File: tests/Mocks/Servers.h [hunk 14]
```diff
@@ -23,6 +23,7 @@
 #ifndef ARANGODB_TESTS_MOCKS_SERVERS_H
 #define ARANGODB_TESTS_MOCKS_SERVERS_H 1
 
+#include <IResearch/AgencyMock.h>
```
Comment (goedderz) [inline tests/Mocks/Servers.h]:
```suggestion
#include "IResearch/AgencyMock.h"
```

File: tests/Mocks/Servers.cpp [hunk 15]
```diff
@@ -85,6 +85,7 @@
 #include <velocypack/Slice.h>
 #include <velocypack/velocypack-aliases.h>
 
+#include <Agency/AsyncAgencyComm.h>
```
Comment (goedderz) [inline tests/Mocks/Servers.cpp]:
```suggestion
#include "Agency/AsyncAgencyComm.h"
```

**Comment by maierlars** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/8190/

**Comment by maierlars** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/8196/

**Comment by maierlars** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/8410/

**Comment by maierlars** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/8417/

**Comment by maierlars** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/8445/

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/8498/

**Comment by maierlars** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/8536/

**Comment by maierlars** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/8553/

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/8784/

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/8792/

**Comment by maierlars** [comment]
http://172.16.10.101:8080/job/arangodb-matrix-pr/8824/

**Comment by maierlars** [comment]
http://172.16.10.101:8080/job/arangodb-matrix-pr/8839/

**Comment by maierlars** [comment]
http://172.16.10.101:8080/job/arangodb-matrix-pr/8840/

**Comment by maierlars** [comment]
http://172.16.10.101:8080/job/arangodb-matrix-pr/8845/

**Comment by maierlars** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/9065/

**Comment by maierlars** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/9403/

**Comment by maierlars** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/9415/

**Comment by maierlars** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/9424/

**Comment by maierlars** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/9425/

**Comment by maierlars** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/9455/

**Comment by maierlars** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr-linux/11213/

**Comment by maierlars** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr-linux/11215/

**Comment by maierlars** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr-linux/11218/

**Comment by maierlars** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr-linux/11224/

**Comment by maierlars** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr-linux/11231/

**Comment by maierlars** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr-linux/11240/

**Comment by maierlars** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr-linux/11241/

**Comment by maierlars** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr-linux/11246/

**Comment by maierlars** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr-linux/11250/

**Comment by maierlars** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr-linux/11252/

**Comment by maierlars** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/9593/

**Comment by maierlars** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/9671/

**Comment by maierlars** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/9725/

**Comment by maierlars** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/9727/

**Comment by neunhoef** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/9733/

**Comment by neunhoef** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr-linux-nonmaintainercompile/9302/

**Comment by neunhoef** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/9739/

**Comment by neunhoef** [comment]
Hopefully final run.

**Pull request merged**